File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s749703010_comments_removed.py
def dp(flip, idx, msk):
    if msk == (1 << n) - 1:
        return 0
    if memo[flip][idx][msk] != -1:
        return memo[flip][idx][msk]
    x = bin(msk).count('1')
    ff = 0
    ans = float('inf')
    for i in range(n):
        if (msk & (1 << i)) == 0:
            nf = x % 2
            if idx == n or a[flip][idx] <= a[(nf + (ff % 2)) % 2][i]:
                ans = min(ans, ff + dp((nf + (ff % 2)) % 2, i, msk | (1 << i)))
            ff += 1
        else:
            x -= 1
    memo[flip][idx][msk] = ans
    return ans

n = int(input())
a = [[0] * n for _ in range(2)]
a[0] = [int(input()) for _ in range(n)]
a[1] = [int(input()) for _ in range(n)]
memo = [[[ -1 for _ in range(1 << n)] for _ in range(n + 1)] for _ in range(2)]
ans = dp(0, n, 0)
print(-1 if ans >= 1e8 else ans)  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s749703010_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

type Pair struct {
	l int
	r int
}

func (p Pair) String() string {
	return fmt.Sprintf("%d %d", p.l, p.r)
}

var memo [2][100][1024]int
var a [2][100]int
var n int

func dp(flip, idx, msk int) int {
	if msk == (1 << n) - 1 {
		return 0
	}
	if memo[flip][idx][msk] != -1 {
		return memo[flip][idx][msk]
	}
	x := bitCount(msk)
	ff := 0
	ans := 1000000000
	for i := 0; i < n; i++ {
		if (msk & (1 << i)) == 0 {
			nf := x % 2
			if idx == n || a[flip][idx] <= a[(nf + (ff % 2)) % 2][i] {
				ans = min(ans, ff + dp((nf + (ff % 2)) % 2, i, msk | (1 << i)))
			}
			ff++
		} else {
			x--
		}
	}
	memo[flip][idx][msk] = ans
	return ans
}

func bitCount(n int) int {
	count := 0
	for n > 0 {
		count += n & 1
		n >>= 1
	}
	return count
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	n = readInt(sc)
	a = [2][100]int{}
	for i := 0; i < n; i++ {
		a[0][i] = readInt(sc)
	}
	for i := 0; i < n; i++ {
		a[1][i] = readInt(sc)
	}
	for i := range memo {
		for j := range memo[i] {
			for k := range memo[i][j] {
				memo[i][j][k] = -1
			}
		}
	}
	ans := dp(0, n, 0)
	if ans >= 100000000 {
		fmt.Println(-1)
	} else {
		fmt.Println(ans)
	}
}

func readInt(sc *bufio.Scanner) int {
	sc.Scan()
	num, _ := strconv.Atoi(sc.Text())
	return num
}
// End of Code
